### DISCUSSION TITLE
Modern Web Apps 2017 - Rotas para Get e Delete

### DISCUSSION THREAD

**Discussion by KIQxl** [balta-io/forum#567] [Tira Dúvidas] [answered]
Boa tarde!
Estava assistindo esse curso após assistir os cursos mais atuais afim de entender um pouco mais sobre arquitetura, domínios ricos e cqrs e acabei me deparando com uma duvida, no curso é mostrado nas controllers da API algumas rotas com método POST relacionadas a criação de clientes e pedidos, nesses casos o fluxo do código é gerenciado por um Handler. No caso das rotas para GET onde eu não tenho corpo para a requisição sendo assim não consigo passar um command para um handler que me retorne os registros do banco, como eu faria nesse caso para o GET? não é indicado ter um handler específico para esse método ou eu faria isso direto pelo repositório na propria controller?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @KIQxl, como vai?

É sim, no caso o Command pode ter apenas o Id do item que deseja buscar por exemplo, ou se for algo mais complexo podemos até testar os parâmetros de entrada e saída.

O curso que você está assistindo é bem antigo, já tem abordagens mais novas no site: 

https://balta.io/player/assistir/5c350f62-e717-9a7d-1241-702a00000000
https://balta.io/player/assistir/5d081362-fb6f-c00e-79ab-8a9b00000000
https://balta.io/player/assistir/650591a8-0795-45aa-9931-85a3d783bf65
https://balta.io/player/assistir/dfa3752e-4bd5-4101-bafb-ff3afcf39930
